base/cpuid.cpp

```diff
@@ -39,6 +41,25 @@ CPUIDResult CPUID(std::uint32_t const eax, std::uint32_t const ecx) {
 
 }  // namespace
 
+CPUIDFeatureFlag::CPUIDFeatureFlag(std::string_view const name,
+                                   std::uint32_t const leaf,
+                                   std::uint32_t const sub_leaf,
+                                   std::uint32_t CPUIDResult::*const field,
+                                   std::int8_t const bit)
+    : name_(name), leaf_(leaf), sub_leaf_(sub_leaf), field_(field), bit_(bit) {
+  CHECK_GE(bit, 0);
+  CHECK_LT(bit, 32);
+  CPUIDFlags().push_back(*this);
+}
+
+std::string_view CPUIDFeatureFlag::name() const {
+  return name_;
+}
+
+bool CPUIDFeatureFlag::IsSet() const {
+  return CPUID(leaf_, sub_leaf_).*field_ & (1 << bit_);
```

Comment:
It would be good if this function was fast.  Why don't we extract the bit at construction instead of computing it on each call?  Also, we should mark this function as inline.

Reply:
We call this once.